### ISSUE TITLE
Https not working

### ISSUE THREAD

**Issue by hurnhu** [giuliohome/WifiHotspotProxy#1]
How can i make this work for https links? seems to work fine for http.

**Comment by giuliohome** [comment]
Hello! No absolutely my phone is not rooted. Ok I'll try to explain what was my objective with this program. I'll reply with a more detailed comment soon. The scenario is that you are supposed have this particular phone acting as a router and two machines at least connected, one of which being a web server and the other a web client. Are you sure this is your scenario? Anyway, more details soon

**Comment by giuliohome** [comment]
Noticed now that you are asking about https. Ok I'll test it, but again we are speaking about connections between internal machines behind the phone router: this is not for https connection to the external internet.
Sorry if it was already understood, but I just wanted to be crystal clear about that.

**Edit**
I've seen your emoji and that you seem disappointed but let me clarify again:
What I mean here is that _for external connections_ your phone should be already acting as a hotspot and no additional app should be needed, so it's not a bad news.

**Comment by hurnhu** [comment]
@giuliohome ive updated my original issue. I really just looping for a http/https proxy. 
I've got it working, but it just doesnt work for https.

What im trying to do is have my phone work as a http/https proxy for my computer while i am using the hotspot on my phone. As using the proxy is a work around to get faster hotspot when they cut my speed becuase I used too much data. 

:( sadly im looking for something for external connections

**Comment by giuliohome** [comment]
If you have your computer connected to the phone and the phone is acting as hotspot, this should be a standard case already supported by the native hotspot of the phone. You shouldn't need any special app. Sorry for the confusion, but what am I missing? Which is exactly the issue with your phone hotspot and why do you think you need to install an app?

**Comment by hurnhu** [comment]
I am able to connect and use my hotspot fine.

But if i use 15GB of data of my hotspot with in a month my carrier slows my connection to 600kbps. 
If i run a app that works as a https/http proxy on my phone while using the hotspot my connection speed go back to >20mbps. Even if i have already used my entire 15GB for the month.

I've tried this with 2 apps and confirmed its working. But i dont trust a proxy made by someone else.

**Comment by giuliohome** [comment]
This app is not doing what you say (it is designed for another usage).
I can try to study what you are asking and I can get back to you here if I find something helpful for you. At the moment I can't promise, also because I'm a bit skeptical: why would your carrier be tricked by a proxy? It should keep track of your downloads irrespectively of any proxy you set. But you confirm you have found 2 app working, so, I don't know... What I'm sure about is that this project currently can't help you to achieve such a trick. 
I will add other details if I have more to say, I don't want to close it before a more in depth search.

**Comment by hurnhu** [comment]
My carrier will give unrestricted download speed if using a proxy. Or by setting `defaultcurhoplimit` to 64 on the PC and using a VPN on the PC. But then im restricted to how every fast the VPN is.

**Comment by giuliohome** [comment]
you have to set the target server (e.g. github.com) in `network-security-config` and then you can test like below 
`curl --proxy http://192.168.43.1:8000 github.com/giuliohome`
The above will fetch the github page via the phone router acting as **https proxy** and return the html page to the curl cmd. I've tested it on my phone 
running the app in debug on intellij idea. Of course it's only a proof of concept, it's not directly usable in a browser (and it's not supposed to be a proxy for the external internet indeed).

**Comment by giuliohome** [comment]
What I have described here is similar to a so called "man in the middle" technique for https. Since my original issue has been solved by Xiaomi now, I've accepted to modify this repository to draft such a https connection, for debugging purposes. Notice however that your intention here was off topic and also very misleading. Try to understand that the bandwidth limitations imposed by a provider can't be overcome through a proxy connection, but at best a proxy can use the available bandwidth as max as possible with some optimization or even compression. That would expose anyway one's privacy to a risk of an unreliable third party store, no matter if the code is open sourced (which is generally a good thing, though).
